flare/rbcm.py

```diff
@@ -0,0 +1,1267 @@
+import json
+import logging
+import pickle
+import time
+
+import multiprocessing as mp
+import numpy as np
+
+from collections import Counter
+from numpy.random import random
+from scipy.optimize import (
+    minimize,
+    basinhopping,
+    differential_evolution,
+    dual_annealing,
+)
+from scipy.cluster.hierarchy import linkage, fcluster
+from typing import List
+
+import matplotlib.pyplot as plt
+
+plt.switch_backend("agg")
+
+from flare.env import AtomicEnvironment
+from flare.gp import GaussianProcess
+from flare.gp_algebra import (
+    get_like_from_mats,
+    get_neg_like_grad,
+    get_ky_mat_update,
+    _global_training_data,
+    _global_training_labels,
+    _global_training_structures,
+    _global_energy_labels,
+    get_Ky_mat,
+    get_kernel_vector,
+    get_neg_like,
+    kernel_distance_mat,
+)
+from flare.kernels.utils import from_mask_to_args
+from flare.output import set_logger
+from flare.parameters import Parameters
+from flare.struc import Structure
+from flare.utils.element_coder import NumpyEncoder, Z_to_element
+from typing import Union
+
+
+class RobustBayesianCommitteeMachine(GaussianProcess):
+    """Robust Bayesian Committee Machine.
+    Splits up prediction and hyperparameter optimzation over a 'comittee'
+    of experts, which gets around the unfavorable nonlinear scaling of
+    preidction and training time with training set size. Particularly for
+    optimizing hyperparameters (typically the most
+    time-consuming individual step of adapting GPs to a new system).
+
+    opt_algorithm GP kwarg supports "dual annealing" and "differential evolution"
+    for the RBCMs.
+
+    See https://arxiv.org/pdf/1806.00720.pdf for a theory background.
```

Comment:
please cite the original paper instead: Deisenroth, M.P., and Ng, J.W. (2015). Distributed Gaussian Processes. 10.

Because we didn't use GRBCM proposed in the newer paper.